dotCMS/src/main/java/com/dotcms/rest/api/v3/contenttype/FieldResource.java

```diff
@@ -55,17 +61,113 @@ public FieldResource() {
+            @Context final HttpServletRequest req)
+            throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData =
+                this.webResource.init(null, true, req, true, null);
+        final User user = initData.getUser();
+
+        final Field fieldToUpdate = updateFieldForm.getField();
+        final ContentType contentType = APILocator.getContentTypeAPI(user).find(typeIdOrVarName);
+
+        checkFieldExists(fieldToUpdate, contentType);
+
+        final FieldLayout fieldLayout = new FieldLayout(contentType.fields());
+        final FieldLayout fieldLayoutUpdated = fieldLayout.update(list(fieldToUpdate));
+
+        fieldLayoutUpdated.validate();
+        fieldAPI.save(fieldToUpdate, user);
+
+        final List<Field> contentTypeFields = fieldAPI.byContentTypeId(contentType.id());
+        final FieldLayout fieldLayoutFromDB = new FieldLayout(contentTypeFields);
+
+        return Response.ok(new ResponseEntityView(fieldLayoutFromDB.getRows())).build();
+    }
+
+
+    /**
+     * Create a new field and return the new {@link ContentType}'s layout in the response
+     *
+     * @param typeIdOrVarName
+     * @param updateFieldForm
+     * @param req
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+    public Response createField(
+            @PathParam("typeIdOrVarName") final String typeIdOrVarName,
+            final UpdateFieldForm updateFieldForm,
+            @Context final HttpServletRequest req)
+            throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData =
+                this.webResource.init(null, true, req, true, null);
+        final User user = initData.getUser();
+
+        final Field fieldToUpdate = updateFieldForm.getField();
+        final ContentType contentType = APILocator.getContentTypeAPI(user).find(typeIdOrVarName);
+
+        final FieldLayout fieldLayout = new FieldLayout(contentType.fields());
+        final FieldLayout fieldLayoutUpdated = fieldLayout.update(list(fieldToUpdate));
+
+        fieldLayoutUpdated.validate();
+        fieldAPI.save(fieldToUpdate, user);
+
+        final List<Field> contentTypeFields = fieldAPI.byContentTypeId(contentType.id());
+        final FieldLayout fieldLayoutFromDB = new FieldLayout(contentTypeFields);
+
+        return Response.ok(new ResponseEntityView(fieldLayoutFromDB.getRows())).build();
+    }
+
+    private void checkFieldExists(Field fieldToUpdate, ContentType contentType) {
+        Optional<Field> optionalField = contentType.fields()
+                .stream()
+                .filter(field -> field.id().equals(fieldToUpdate.id()))
+                .findFirst();
+
+        if (!optionalField.isPresent()) {
+            throw new NotFoundException("Field does not exists");
```

Comment:
Please add the Velocity var name or any information of the field that doesn't exist, for troubleshooting purposes.

Reply:
I agree